tests/unit_tests/main_test.cpp

```diff
@@ -116,7 +116,11 @@ int main(int argc, char** argv) {
                         if (dev.is_gpu() && vendor_id == NVIDIA_ID)
                             continue;
 #endif
+#ifdef __HIPSYCL__
```

Comment:
What is the reason for this change?

Reply:
I think this is about a difference in classification of devices between hipSYCL and DPC++. The SYCL specification does not really define what "accelerator" is and when `is_accelerator()` should return true. In particular, it does not say that `is_accelerator()` and `is_gpu()` should be mutually exclusive. In hipSYCL, for a GPU both `is_gpu() == true` as well as `is_accelerator() == true`. This change therefore is required to allow selecting a  GPU in hipSYCL.

If we want to exclude FPGAs, maybe we could generalize it to `if(dev.is_gpu() || !dev.is_accelerator())`?

Reply:
@illuhad Thanks for the explanation. I agree with you. I also see same behavior in my local tests. I am ok with the current implementation.